uefi/region_test.go

```diff
@@ -0,0 +1,61 @@
+package uefi
+
+import "testing"
+
+var regionTestcases = [...]struct {
+	in    *Region
+	valid bool
+	base  uint32
+	end   uint32
+}{
+	// Invalid
+	{
+		in:    &Region{0, 0},
+		valid: false,
+		base:  0,
+		end:   0x1000,
+	},
+	{
+		in:    &Region{1, 0},
+		valid: false,
+		base:  0x1000,
+		end:   0x1000,
+	},
+	// Valid
+	{
+		in:    &Region{1, 1},
+		valid: true,
+		base:  0x1000,
+		end:   0x2000,
+	},
+	{
+		in:    &Region{100, 200},
+		valid: true,
+		base:  0x64000,
+		end:   0xC9000,
+	},
+}
+
+func TestRegion_Valid(t *testing.T) {
+	for _, tc := range regionTestcases {
+		if out := (*Region).Valid(tc.in); out != tc.valid {
```

Comment:
Can't you do tc.in.Valid() here?

Reply:
Likewise for all other instances of (*Region).Valid

Reply:
I think using `(*Region).Valid(%#v)` makes sense in the error strings. This way I'm consistent with that.

Let me know if you still want it to change.

Reply:
I think this is confusing though. In the error strings I'd much rather something like:

t.Errorf("Region %v.Valid() = %v, expected %v", *tc.in, out, tc.valid)